src/main/java/Duke/DukeSave.java

```diff
@@ -0,0 +1,96 @@
+package Duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+//import java.nio.file.Files;
+//import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class DukeSave {
+    /**
+     * Divider used to separate details of a task saved in file
+     */
+    private static final String SAVEFILESEPARATOR = ", ";
+    private static String filePath = "DukeTaskList.txt";
+    private static File textFile;
+    private static Scanner textFileScanner;
+
+    private static void setupFile() {
+        textFile = new File(filePath);
+
+        try {
+            textFileScanner = new Scanner(textFile);
+        } catch (FileNotFoundException e) {
+            System.out.println("I cannot seem to access the saved tasks. Did you perhaps lock it away?");
+        }
+    }
+    
+    public static void saveDukeToFile(ArrayList<DukeTasks> list) {
+        String writeToFile = "";
+        
+        for (DukeTasks dukeTasks : list) {
+            // extracting details from task object
+            String[] dataToTextFile= {dukeTasks.getTaskType(), dukeTasks.isDone.toString(), dukeTasks.description};
+            if (dukeTasks instanceof DukeDeadlines) {
+                dataToTextFile[2] += "/by" + ((DukeDeadlines) dukeTasks).getBy();
```

Comment:
Magic strings for `"/by"` here. There are also similar magic strings/literals/numbers issues in this and other files. Do take a look at them when you have time.